lua-tree/share/lua/5.1/kong/db/schema/entities/routes.lua

```diff
@@ -42,23 +40,6 @@ if kong_router_flavor == "expressions" then
       { expression = { type = "string", required = true }, },
       { priority = { type = "integer", required = true, default = 0 }, },
     },
-
-    entity_checks = {
-      { custom_entity_check = {
-        field_sources = { "expression", "id", },
-        fn = function(entity)
-          local s = atc.get_schema()
-          local r = router.new(s)
-
-          local res, err = r:add_matcher(0, entity.id, entity.expression)
-          if not res then
-            return nil, "Router Expression failed validation: " .. err
-          end
-
-          return true
-        end,
-      } },
-    },
```

Comment:
I am on the fence about this change. I understand we need this removal and the requires due to them not being available in gopher-lua, but WDYT about this instead.

1. Move the local imports into the condition for `expression`
2. Force the `kong_router_flavor` to be traditional and add a comment
3. Revert the removal of the entity_checks here

```
-- ATC router is incompatible with goks
local kong_router_flavor = "traditional"

if kong_router_flavor == "expressions" then
  local atc = require("kong.router.atc")
  local router = require("resty.router.router")
  return {
```

My reasoning here is less patch and if the defaults change out from underneath us and we are not ready then this catch will ensure nothing breaks.

Reply:
`local r = router.new(s)` uses the FFI interface (due to it using [atc-router](https://github.com/Kong/atc-router)), importing `resty.router.router` wouldn't work as intended as such. Am I missing something?

Reply:
Agreed it wouldn't work, but it should never hit that condition which means those ATC imports will never get performed either. We could also just remove everything in here for the ATC router, remove the `if` block and keep only the `else` block. WDYT?

Reply:
Ah, I see. That brings up another comment then, the "expressions" route typedef & "traditional" route typedef are not compatible with one another. In the event someone were using a custom plugin schema that took advantage of the newer route typedef, they would be unable to use fields like "priority".